ui/apps/platform/src/Containers/ComplianceEnhanced/ScanConfigs/Wizard/ReviewConfig.tsx

```diff
@@ -1,9 +1,93 @@
-import React, { ReactElement } from 'react';
+import React from 'react';
+import { FormikContextType, useFormikContext } from 'formik';
+import {
+    Divider,
+    Flex,
+    FlexItem,
+    PageSection,
+    Stack,
+    StackItem,
+    Text,
+    TextList,
+    TextListItem,
+    TextVariants,
+    Title,
+} from '@patternfly/react-core';
+
+import { ClusterScopeObject } from 'services/RolesService';
+import { ComplianceProfile } from 'services/ComplianceEnhancedService';
+
+import { ScanConfigFormValues } from './useFormikScanConfig';
+import {
+    convertFormikParametersToSchedule,
+    formatScanSchedule,
+} from '../compliance.scanConfigs.utils';
+
+export type ProfileSelectionProps = {
+    clusters: ClusterScopeObject[];
+    profiles: ComplianceProfile[];
+};
+
+function ReviewConfig({ clusters, profiles }: ProfileSelectionProps) {
+    const { values: formikValues }: FormikContextType<ScanConfigFormValues> = useFormikContext();
+
+    const scanSchedule = convertFormikParametersToSchedule(formikValues.parameters);
+    const formattedScanSchedule = formatScanSchedule(scanSchedule);
+
+    function findById<T extends { id: string }>(selectedIds: string[], items: T[]): T[] {
+        return selectedIds
+            .map((id) => items.find((item) => item.id === id))
+            .filter((item): item is T => item !== undefined);
+    }
+
+    const selectedClusters = findById<ClusterScopeObject>(formikValues.clusters, clusters);
+    const selectedProfiles = findById<ComplianceProfile>(formikValues.profiles, profiles);
```

Comment:
This is good, and might be worthwhile to move to a global utility in the future. 👍 I feel like I've had to do something similar in the past.

I bet with the way this is written that you could even omit the generic type in the function calls and TS would infer it correctly.

Reply:
Super that `findById` function correctly handles situation if no item has an `id` so am curious to confirm my understanding that is not expected in this situation of selecting clusters and profiles.

Reply:
That's a good idea, but I'm unfortunately going to have to modify this code once I hookup to the correct api. I missed the fact I'll be using `cluster.clusterId` as opposed to `cluster.id`.